File: simple-stack-example-mvvm-fragments/src/main/java/com/example/android/architecture/blueprints/todoapp/application/BaseFragment.java
package com.example.android.architecture.blueprints.todoapp.application;

import android.support.v4.app.Fragment;

/**
 * Created by Zhuinden on 2017.07.26..
 */

public class BaseFragment
        extends Fragment {
    public final <T extends BaseKey> T getKey() {
        return getArguments() != null ? getArguments().getParcelable("KEY") : null;
    }
}

File: simple-stack-example-scoping/src/main/kotlin/com/zhuinden/simplestackexamplescoping/BaseFragment.kt
package com.zhuinden.simplestackexamplescoping

import android.support.v4.app.Fragment

/**
 * Created by Zhuinden on 2018.09.17.
 */

open class BaseFragment : Fragment() {
    fun <T : BaseKey> getKey(): T = requireArguments.getParcelable<T>("KEY")
}
